protoc-gen-grpc-gateway/gengateway/generator.go

```diff
@@ -103,7 +102,7 @@ func (g *generator) generate(file *descriptor.File) (string, error) {
 	for _, svc := range file.Services {
 		for _, m := range svc.Methods {
 			pkg := m.RequestType.File.GoPkg
-			if m.Options == nil || !proto.HasExtension(m.Options, options.E_Http) ||
+			if len(m.Bindings) == 0 ||
```

Comment:
Can you help me understand why this is preferential? Could you also add a test demonstrating why this is preferred

Reply:
This is from 3356bb1 :

> Previously the generator generator checked whether the http extension
> was used to decide whether imports were required for a method. This
> patch makes it check whether a method has bindings instead as those
> are what code is actually generated for.

Basically the previous check skipped adding imports when the protobuf file wasn't using the Http annotation. That check is done on data protoc passed from the proto file. Since the whole point of this PR is to no longer have the Http annotations in the proto this check had to be amended. Now it checks whether we created bindings for the method (it should have a binding for each endpoint). This will happen based on annotation or the yaml.

What you see without this change is that the generated .gw.go file might be missing imports (e.g. if the method uses ```google.protobuf.Duration``` for which an import is required). In that case it just won't compile.

I will see if I can do a test for that.